be/src/Project/Services/DownloadCenter/DownloadCenter.Api/Services/DownloadCenter/DownloadCenterService.cs

```diff
@@ -0,0 +1,348 @@
+
+namespace DownloadCenter.Api.Services.DownloadCenter
+{
+    public class DownloadCenterService : IDownloadCenterService
+    {
+        private readonly DownloadCenterContext context;
+        private readonly IStringLocalizer<DownloadCenterResources> downloadCenterLocalizer;
+
+        public DownloadCenterService(
+            IStringLocalizer<DownloadCenterResources> downloadCenterLocalizer,
+            DownloadCenterContext context)
+        {
+            this.context = context;
+            this.downloadCenterLocalizer = downloadCenterLocalizer;
+        }
+
+        public async Task<Guid> CreateAsync(CreateDownloadCenterFileServiceModel model)
+        {
+            foreach(var categoryId in model.CategoriesIds.OrEmptyIfNull())
+            {
+                foreach(var file in model.Files.OrEmptyIfNull())
+                {
+                    var categoryFile = new CategoryFile
+                    {
+                        CategoryId = categoryId,
+                        MediaId = file.Id,
+                        Name = file.Name
+                    };
+
+                    await this.context.CategoryFiles.AddAsync(categoryFile.FillCommonProperties());
+                }
+            }
+
+            await this.context.SaveChangesAsync();
+
+            return model.Files.FirstOrDefault().Id;
+        }
+
+        public async Task DeleteAsync(DeleteDownloadCenterFileServiceModel model)
+        {
+            var downloadCenterFile = this.context.CategoryFiles.Where(x => x.MediaId == model.Id && x.IsActive);
+
+            if (downloadCenterFile is null)
+            {
+                throw new CustomException(this.downloadCenterLocalizer.GetString("DownloadCenterFileNotFound"), (int)HttpStatusCode.NotFound);
+            }
+
+            foreach(var downloadCenterCategoryFile in downloadCenterFile.OrEmptyIfNull())
+            {
+                downloadCenterCategoryFile.IsActive = false;
+                downloadCenterCategoryFile.LastModifiedDate = DateTime.UtcNow;
+            }
+
+            await this.context.SaveChangesAsync();
+        }
+
+        public async Task<PagedResults<IEnumerable<DownloadCenterFileServiceModel>>> GetAsync(GetDownloadCenterFilesServiceModel model)
+        {
+            var downloadCenterFiles = this.context.CategoryFiles.Where(x => x.IsActive);
+
+            if (string.IsNullOrWhiteSpace(model.SearchTerm) is false)
+            {
+                var searchFiles = downloadCenterFiles.Where(x => x.Id.ToString() == model.SearchTerm || x.Name.StartsWith(model.SearchTerm));
+
+                var categoryTranslation = await this.context.CategoryTranslations.FirstOrDefaultAsync(x => x.Name.StartsWith(model.SearchTerm));
+
+                if (categoryTranslation is not null)
+                {
+                    searchFiles = downloadCenterFiles.Where(x => x.CategoryId == categoryTranslation.CategoryId);
+                }
+
+                downloadCenterFiles = searchFiles;
```

Comment:
It shouldn't be inactive because the isActive flag is on line 75.

I know it is overwriting. That's what I meant because if the category was not found, the result was always null despite looking for a filename.

If I'm wrong, let me know